packages/debug/package.json

```diff
@@ -19,6 +19,7 @@
     "jsonc-parser": "^2.0.2",
     "mkdirp": "^0.5.0",
     "p-debounce": "^1.0.0",
+    "requestretry": "^3.1.0",
```

Comment:
Does this dependency addition attract an Eclipse CQ?

Reply:
@marcdumais-work Do you have details on https://github.com/theia-ide/theia/wiki/Registering-CQs#case-new-or-updated-production-dependency?

Reply:
Or it is a case for https://github.com/theia-ide/theia/wiki/Registering-CQs#case-new-or-updated-testbuild-dependency and we don't need CQ? This dependency will be installed by clients but only to download debug adapters, not in production code.

Reply:
This may be dubious as you need it in devDependencies even though its a developer dependency?

Reply:
Here is a CQ: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=18783

Does it look good?

Reply:
LGTM, @marcdumais-work is the authority, however :)

Reply:
> This dependency will be installed by clients but only to download debug adapters, not for production code.

As Florent says, if it's in the "Dependencies", it's considered a production/runtime dep. It doesn't really matter how it's used. if we "link" to it. 

> Here is a CQ: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=18783
> Does it look good?

I do not think so. 

For production/runtime dependencies, we need to consider not only that dependency but the recursive tree of rutime-dependencies-of-dependencies as well. The suggested way to do that so far is to submit the whole archive of the project's dependencies (node_modules folder). I have submitted such a CQ last week that's still being analysed. We would need to update the included archive/zip so that it covers this new dependency, recursively: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=18717

I have notes how I obtained the "node_modules" folder I attached to CQ #18717.  @akosyakov you want to give it a try, and attach an updated .zip to the existing CQ? This might show that the current rules do not scale so well...

Reply:
@akosyakov I have updated the instructions here: https://github.com/theia-ide/theia/wiki/Registering-CQs#case-new-or-updated-production-dependency

Looking quickly at this PR, I think we will need to update `DAlist` so we point to the new folders where the Debug Adapters are saved on disk.

Reply:
> Looking quickly at this PR, I think we will need to update DAlist so we point to the new folders where the Debug Adapters are saved on disk.

@marcdumais-work i'm not sure what you mean. I've closed another PR with bundling vsix archives. This PR tries to redownload archives several times, nothing is bundled together with the extension.

> I have notes how I obtained the archive I attached to CQ #18717. @akosyakov you want to give it a try, and attach an updated .zip to the existing CQ? This might show that the current rules do not scale so well...

I want to have the stable build 😢 

[Looking at yarn.lock](https://github.com/theia-ide/theia/pull/4120/files#diff-8ee2343978836a779dc9f8d6b794c3b2) there are 2 new dependencies: `requestretry@3.1.0` and `when@3.7.8`. I've opened a CQ only for `requestretry`. Would be opening an additional CQ for `when` enough?

Reply:
a CQ for `when` package: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=18790

Reply:
update: CQ https://dev.eclipse.org/ipzilla/show_bug.cgi?id=18717 now covers the new prod deps in this PR.

Reply:
@marcdumais-work as far as i understood we agreed to not hold PRs because of such CQs yet? Could it be updated and merged? download script is still failing silent some times and does not always work for clients (besides travis)

Reply:
@akosyakov We have not yet agreed yet, but in this case I would go ahead and merge; CQ #18717 is blocked again (newfound license issue not related to this PR here) and might remain so for a while.